android/src/com/mapswithme/maps/routing/RoutingPlanController.java

```diff
@@ -108,18 +127,29 @@ public void onClick(View v)
       }
     });
 
+    setupRouterButton(R.id.taxi, R.drawable.ic_taxi, new View.OnClickListener()
+    {
+      @Override
+      public void onClick(View v)
+      {
+        AlohaHelper.logClick(AlohaHelper.ROUTING_TAXI_SET);
```

Comment:
Probably it's better but I'm not sure. I just did it the same as it was before - http://clip2net.com/s/3DcIqK7 . I wouldn't touch this logic in scope of this PR. BTW, I guess this logic should be in Java code, because it's supposed to be called from main thread (I'm not sure)